src/network/ctf_client_list_handling_system.cpp

```diff
@@ -0,0 +1,85 @@
+#include "ctf_client_list_handling_system.h"
+#include "platform/event.h"
+#include "team_switch_request_event.h"
+#include "ctf_client_datas_changed_event.h"
+#include "client_ready_event.h"
+#include <algorithm>
+
+namespace network {
+
+
+CtfClientListHandlingSystem::CtfClientListHandlingSystem()
+{
+    mOnTeamSwitchRequestEvent = 
+        platform::EventServer<network::TeamSwitchRequestEvent>::Get().Subscribe( boost::bind( &CtfClientListHandlingSystem::OnTeamSwitchRequestEvent, this, _1 ) );
+    mClientReadyEvent = 
+        platform::EventServer<network::ClientReadyEvent>::Get().Subscribe( boost::bind( &CtfClientListHandlingSystem::OnClientReadyEvent, this, _1 ) );
+}
+
+void CtfClientListHandlingSystem::Init()
+{
+}
+
+void CtfClientListHandlingSystem::Update( double DeltaTime )
+{
+}
+
+void CtfClientListHandlingSystem::OnTeamSwitchRequestEvent( TeamSwitchRequestEvent const& event )
+{
+    std::vector<::ctf::ClientData>::iterator it = std::find_if( mClientDatas.begin(), mClientDatas.end(),
+           boost::bind<bool>( []( ::ctf::ClientData const& d, int32_t clientId ){ return d.mClientId == clientId; }, _1, event.mClientId ) );
+    if ( mClientDatas.end() == it )
+    {
+        // something went wrong if we end up here
+        return;
+    }
+    Team::Type team = (*it).mTeam;
+    int teamsize = std::count_if( mClientDatas.begin(), mClientDatas.end(),
+           boost::bind<bool>( []( ::ctf::ClientData const& d, Team::Type t ){ return d.mTeam == t; }, _1, team ) );
+
+    // don't allow switch in team of size 1 -> it would become empty
+    if ( teamsize > 1 )
```

Comment:
yes, I had doubts about it. I just wanted to avoid the case to allow empty team creation even accidentally. I'll remove it we'll see how good it is